src/clients/SpokePoolClient.ts

```diff
@@ -1,4 +1,225 @@
+      this.logger.debug({ at: "SpokePoolClient#init", message: `Listening for ${this.chain} events.` });
+    }
+  }
+
+  /**
+   * Receive an update from the external indexer process.
+   * @param rawMessage Message to be parsed.
+   * @returns void
+   */
+  protected indexerUpdate(rawMessage: unknown): void {
+    assert(typeof rawMessage === "string", `Unexpected ${this.chain} message data type`);
+
+    const message = JSON.parse(rawMessage);
+    if (isSpokePoolEventRemoved(message)) {
+      const event = JSON.parse(message.event, sdkUtils.jsonReviverWithBigNumbers);
+      this.pendingEventsRemoved.push(event);
+      return;
+    }
+
+    assert(isSpokePoolEventsAdded(message), `Expected ${this.chain} SpokePoolEventsAdded message`);
+
+    const { blockNumber, currentTime, oldestTime, nEvents, data } = message;
+    if (nEvents > 0) {
+      const pendingEvents = JSON.parse(data, sdkUtils.jsonReviverWithBigNumbers);
+      assert(
+        Array.isArray(pendingEvents) && pendingEvents.length === nEvents,
+        Array.isArray(pendingEvents)
+          ? `Expected ${nEvents} ${this.chain} pendingEvents, got ${pendingEvents.length}`
+          : `Expected array of ${this.chain} pendingEvents`
+      );
+
+      this.logger.debug({
+        at: "SpokePoolClient#indexerUpdate",
+        message: `Received ${nEvents} ${this.chain} events from indexer.`,
+      });
+
+      pendingEvents.forEach((event) => {
+        const eventIdx = this.queryableEventNames.indexOf(event.event);
+        assert(
+          eventIdx !== -1 && event.removed === false,
+          event.removed ? "Incorrectly received removed event" : `Unsupported event name (${event.event})`
+        );
+
+        this.pendingEvents[eventIdx].push(event);
+      });
+    }
+
+    this.pendingBlockNumber = blockNumber;
+    this.pendingCurrentTime = currentTime;
+    if (!isDefined(this.pendingOldestTime) && oldestTime > 0) {
+      this.pendingOldestTime = oldestTime;
+    }
+  }
+
+  /**
+   * Given an event to be removed, ensure that it is removed from the set of ingested events.
+   * @param event An Ethers event instance.
+   * @returns void
+   */
+  protected removeEvent(event: Event): boolean {
+    let removed = false;
+    const eventIdx = this.queryableEventNames.indexOf(event.event);
+    const pendingEvents = this.pendingEvents[eventIdx];
+
+    const { event: eventName, blockNumber, blockHash, transactionHash, transactionIndex, logIndex } = event;
+
+    // First check for removal from any pending events.
+    const pendingEvent = pendingEvents
+      .map((event, idx) => ({ ...event, idx }))
+      .find((pending) => {
+        pending.logIndex === logIndex &&
+          pending.transactionIndex === transactionIndex &&
+          pending.transactionHash === transactionHash &&
+          pending.blockHash === blockHash;
+      });
+
+    if (isDefined(pendingEvent)) {
+      removed = true;
+
+      // Splice out the events in reverse order.
```

Comment:
Comment is wrong, since this is only splicing a single entry out.
```suggestion
      // Drop the relevant event.
```